linq4j/src/main/java/org/apache/calcite/linq4j/JoinType.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.linq4j;
+
+/**
+ * Enumeration of join types.
+ */
+public enum JoinType {
```

Comment:
Do we want to have two kinds of join types in Linq4j given that we already have CorrelateJoinType? If we didn't have RIGHT join type in this enumeration then the only difference between CorrelateJoinType would be the FULL join. Wouldn't it make sense to deprecate CorrelateJoinType and use this new JoinType everywhere in Linq4j?

Reply:
I agree, I think we should deprecate CorrelateJoinType and use the new JoinType everywhere in Linq4j (although we would need to add some checks to verify that the places expecting a "correlate" type do not get an illegal value: RIGHT / FULL).
IMHO this refactoring should be done in a separate patch.